s3-plugin-server/src/main/scala/com/gu/teamcity/ArtifactUploader.scala

```diff
@@ -31,13 +33,21 @@ class ArtifactUploader(config: S3ConfigManager, s3: S3) extends BuildServerAdapt
                   report("Not configured for uploading")
                   Continuation.BREAK
                 }
-            } recover {
+            } recoverWith {
+              case e: ResetException => {
+                runningBuild.addBuildMessage(new BuildMessage1(DefaultMessagesInfo.SOURCE_ID,
+                  DefaultMessagesInfo.MSG_BUILD_PROBLEM, Status.WARNING, new Date,
+                  s"Retrying artifact upload after error: ${e.getMessage}"))
+
+                s3.upload(config.artifactBucket, runningBuild, buildArtifact.getName,
+                  buildArtifact.getInputStream, buildArtifact.getSize).map(_ => Continuation.CONTINUE)
```

Comment:
I figure if it's already failed once, and then fails again, there's probably little we're going to glean from it.